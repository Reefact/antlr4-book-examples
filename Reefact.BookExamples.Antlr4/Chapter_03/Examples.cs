#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using NFluent;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_03 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        public void convert_a_short_array_to_unicode_string() {
            // Setup
            GRun grun = GRun.ReadString("{99,3,451}");
            // Exercise
            string unicodeString = grun.ToUnicodeString();
            // Verify
            Check.That(unicodeString).IsEqualTo("\"\\u0063\\u0003\\u01C3\"");
        }

        [Fact]
        public void provide_a_lisp_tree_representation_of_a_short_array_in_ArrayInit_grammar() {
            // Setup
            GRun grun = GRun.ReadString("{99,3,451}");
            // Exercise
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Check.That(lispStyleTree).IsEqualTo("(init { (value 99) , (value 3) , (value 451) })");
        }

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void provide_mermaid_style_tree_representation_of_a_short_array_in_ArrayInit_grammar_for_nested_arrays() {
            // Setup
            GRun grun = GRun.ReadString("{1,{2,3},4}");
            // Exercise
            string parseTree1 = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(parseTree1);
        }

    }

}