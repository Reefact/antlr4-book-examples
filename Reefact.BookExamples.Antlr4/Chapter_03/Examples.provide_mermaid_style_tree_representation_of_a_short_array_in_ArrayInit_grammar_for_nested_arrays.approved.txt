graph TD
	1["init"] --> 2["{"]
	1 --> 3["value"]
	3 --> 4["1"]
	1 --> 5[","]
	1 --> 6["value"]
	6 --> 7["init"]
	7 --> 8["{"]
	7 --> 9["value"]
	9 --> 10["2"]
	7 --> 11[","]
	7 --> 12["value"]
	12 --> 13["3"]
	7 --> 14["}"]
	1 --> 15[","]
	1 --> 16["value"]
	16 --> 17["4"]
	1 --> 18["}"]
